src/NuGet.Core/NuGet.CommandLine.XPlat/ReportRenderers/Models/ListPackageProjectModel.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.ProjectModel;
+
+namespace NuGet.CommandLine.XPlat
+{
+    /// <summary>
+    /// Calculated project data model for list report
+    /// </summary>
+    internal class ListPackageProjectModel
+    {
+        private const string ProjectNameMSbuildProperty = "MSBuildProjectName";
+        internal List<ReportProblem> ProjectProblems { get; } = new List<ReportProblem>();
+        internal string ProjectPath { get; private set; }
+        // Original packages
+        internal IEnumerable<FrameworkPackages> Packages { get; private set; }
+        // Calculated project model data for each targetframeworks
+        internal List<ListPackageReportFrameworkPackage> TargetFrameworkPackages { get; private set; }
+        internal string HttpSourceWarning { get; private set; }
+        internal string ProjectName { get; private set; }
+        internal Project Project { get; }
+        internal ListPackageReportModel ReportModel { get; }
+        internal bool AutoReferenceFound { get; private set; }
+
+        public ListPackageProjectModel(string projectPath, ListPackageReportModel reportModel, Project project)
+        {
+            ProjectPath = projectPath;
+            ReportModel = reportModel;
+            Project = project;
+            ProjectName = Project.GetPropertyValue(ProjectNameMSbuildProperty);
+        }
+
+        // For testing purposes only
+        internal ListPackageProjectModel(string projectPath)
+        {
+            ProjectPath = projectPath;
+        }
+
+        internal void SetFrameworkPackageMetadata(List<ListPackageReportFrameworkPackage> frameworkPackages)
+        {
+            TargetFrameworkPackages = frameworkPackages;
+        }
```

Comment:
This isn't Java. Why not make `TargetFrameworkPackages` public/internal settable, rather than private?

Reply:
You're right, addressed.